pmd-core/src/main/java/net/sourceforge/pmd/internal/util/AssertionUtil.java

```diff
@@ -128,4 +137,15 @@ public static RuntimeException mustBe(String name, Object value, String conditio
                                                : prefix + ": " + message;
         return new AssertionError(message);
     }
+
+    public static ContextedRuntimeException addContextValue(RuntimeException e, String label, Object value) {
+        return e instanceof ContextedRuntimeException
+               ? ((ContextedRuntimeException) e).addContextValue(label, value)
+               : new ContextedRuntimeException(e).addContextValue(label, value);
```

Comment:
this would be nicer with a `ContextedRuntimeException.wrap(e)` as you did with `ContextedAssertionError`, wouldn't it?